### DISCUSSION TITLE
Coolify self-hosting guide

### DISCUSSION THREAD

**Discussion by korridor** [solidtime-io/solidtime#252] [Feature requests]
Add solidtime to [Coolify](https://coolify.io/).

**Comment by erykkruk** [comment]
Does someone do it? I ran it, but it's not available :/ 

and I have this issue in app:
`[2025-07-06 17:08:44] production.ERROR: Unable to write to process ID file. {"exception":"[object] (RuntimeException(code: 0): Unable to write to process ID file. at /var/www/html/vendor/laravel/octane/src/FrankenPhp/ServerStateFile.php:49)
[stacktrace]
#0 /var/www/html/vendor/laravel/octane/src/Commands/StartFrankenPhpCommand.php(210): Laravel\\Octane\\FrankenPhp\\ServerStateFile->writeState()
#1 /var/www/html/vendor/laravel/octane/src/Commands/StartFrankenPhpCommand.php(75): Laravel\\Octane\\Commands\\StartFrankenPhpCommand->writeServerStateFile()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Laravel\\Octane\\Commands\\StartFrankenPhpCommand->handle()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(43): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(95): Illuminate\\Container\\Util::unwrapIfClosure()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(696): Illuminate\\Container\\BoundMethod::call()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(213): Illuminate\\Container\\Container->call()
#8 /var/www/html/vendor/symfony/console/Command/Command.php(279): Illuminate\\Console\\Command->execute()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Symfony\\Component\\Console\\Command\\Command->run()
`

**Reply by korridor** [reply]
Are you working on a solidtime Coolify template? If not please post problem with your self-hosting setup in the Discord channel or in a GitHub issue if you suspect that there is a bug.

Regarding you stack trace, please make sure that the folder `storage/logs` exists and is writeable. But to have a definitive answer to your problem I need much more information then just a stack trace.

**Reply by erykkruk** [reply]
Solidtime coolify template exists? 
I'm working on https://github.com/solidtime-io/self-hosting-examples 

On coolify I'm not able to give permissions writeable, do you have any ideas?  Or I just don't know how do it.

**Reply by korridor** [reply]
No a solidtime coolify template doesn't yet exist.
My question was why are you posting the problem that you have self-hosting solidtime in the comment section of a feature request?
I guess you did that because you are trying to run our self-hosting example on coolify, right?

In that case please provide further information on what you did exactly to get to the exception in the logs. What did you configure in coolify? Did you adjust the `docker-compose.yml` file?

**Reply by m-a-mohsen** [reply]
I am using a slightly modified version which I tested today on Coolify
[https://github.com/m-a-mohsen/solidtime-docker/blob/coolify-v01/docker-compose.yml](https://github.com/m-a-mohsen/solidtime-docker/blob/coolify-v01/docker-compose.yml)
env:
[https://github.com/m-a-mohsen/solidtime-docker/blob/coolify-v01/1-docker-with-database/.env.example](https://github.com/m-a-mohsen/solidtime-docker/blob/coolify-v01/1-docker-with-database/.env.example)

The only issue was later with keys
